### ISSUE TITLE
The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

### ISSUE THREAD

**Issue by Jaycyn** [firebase/firebase-ios-sdk#11507]
### Description

When building a fresh macOS Firestore project using Cocoapods, after setting up the project and running it, multiple warnings are displayed across most of the Firestore pods.

The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

### Reproducing the issue

1) Create a fresh macOS project in XCode (This project was called FirestoreTest), set Minimum Deployments to 11.0, App Sandbox Network Incoming and Outgoing checked and remove the Hardened Runtime (not needed for this example)

2) Follow the installation guide here [Install Firestore Apple](https://firebase.google.com/docs/ios/installation-methods?hl=en&authuser=0&_gl=1*wci8ui*_ga*Mjk1MzI5MTM3LjE2ODc0NjMzMzA.*_ga_CW55HF8NVT*MTY4ODQ3NTA1Ny40LjEuMTY4ODQ3NTMyMy4wLjAuMA..#analytics-not-enabled) ensuring that the XCode project is added to the Firestore project via the Firebase Console and the GoogleService-Info.plist downloaded and added to the XCode project

3) Create a podfile (see podfile info below)

4) Quit XCode

5) Command line, cd to the folder containing the project

6)
`pod install --repo-update`

7) Open XCode using the FirestoreTest.xcworkspace file and build

8) Note warnings

### Firebase SDK Version

10.11.0

### Xcode Version

14.3.1

### Installation Method

CocoaPods

### Firebase Product(s)

Authentication, Firestore, Storage

### Targeted Platforms

macOS

### Relevant Log Output

```shell
BoringSSL-GRPC
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase.

FirebaseAppCheckInterop
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseAuth
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

.../FirestoreTest/Pods/FirebaseAuth/FirebaseAuth/Sources/Auth/FIRAuthGlobalWorkQueue.m
.../FirestoreTest/Pods/FirebaseAuth/FirebaseAuth/Sources/Auth/FIRAuthGlobalWorkQueue.m:21:40 A function declaration without a prototype is deprecated in all versions of C

FirebaseAuthInterop
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseCore
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseCoreDiagnostics
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseCoreExtension
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseCoreInternal
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseFirestore
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

.../FirestoreTest/Pods/FirebaseFirestore/Firestore/core/src/util/set_util.h
.../FirestoreTest/Pods/FirebaseFirestore/Firestore/core/src/util/set_util.h:39:11 Parameter 'comparator' not found in the function declaration

.../FirestoreTest/Pods/FirebaseFirestore/Firestore/core/include/firebase/firestore/timestamp.h
.../FirestoreTest/Pods/FirebaseFirestore/Firestore/core/include/firebase/firestore/timestamp.h:106:36 Empty paragraph passed to '@param' command

.../FirestoreTest/Pods/FirebaseFirestore/Firestore/core/include/firebase/firestore/timestamp.h:127:22 Empty paragraph passed to '@param' command

FirebaseFirestoreSwift
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseStorage
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

FirebaseStorageInternal
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

GTMSessionFetcher
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

GoogleDataTransport
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

GoogleUtilities
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.12, but the range of supported deployment target versions is 10.13 to 13.3.99.

Libuv-gRPC
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase.

PromisesObjC
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

abseil
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase.

gRPC-C++
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase.

gRPC-C++-gRPCCertificates-Cpp
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

gRPC-Core
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

Run script build phase 'Create Symlinks to Header Folders' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase.

nanopb
.../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.9, but the range of supported deployment target versions is 10.13 to 13.3.99.
```

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

project 'FirestoreTest.xcodeproj'
platform :osx, '11.0'
target 'FirestoreTest' do
  use_frameworks! :linkage => :static   #required for successful install. Can also `use_modular_headers!`?
  pod 'FirebaseAuth'
  pod 'FirebaseFirestore'
  pod 'FirebaseStorage'
  pod 'FirebaseFirestoreSwift', '~> 9.0.0'
end

```

</details>

**Comment by andrewheard** [comment]
Hi @Jaycyn, would you mind double checking that you are installing Firebase 10.11.0? You can check the console output during `pod install` or inspect the `Podfile.lock` (rather than `Podfile`) file.

My suspicion is that the line `pod 'FirebaseFirestoreSwift', '~> 9.0.0'` in your `Podfile` is resulting in you getting Firebase 9.6.0, the latest 9.x release, rather than 10.11.0. I changed this line to just `pod 'FirebaseFirestoreSwift'`, ran `pod update`, and saw the following:
```
Installing FirebaseAppCheckInterop 10.11.0 (was 9.6.0)
Installing FirebaseAuth 10.11.0 (was 9.6.0)
Installing FirebaseAuthInterop 10.11.0 (was 9.6.0)
Installing FirebaseCore 10.11.0 (was 9.6.0)
Installing FirebaseCoreExtension 10.11.0 (was 9.6.0)
Installing FirebaseCoreInternal 10.11.0 (was 9.6.0)
Installing FirebaseFirestore 10.11.0 (was 9.6.0)
Installing FirebaseFirestoreSwift 10.11.0 (was 9.0.0)
Installing FirebaseSharedSwift (10.11.0)
Installing FirebaseStorage 10.11.0 (was 9.6.0)
Installing GTMSessionFetcher 3.1.1 (was 2.3.0)
Installing abseil 1.20220623.0 (was 1.20211102.0)
Installing gRPC-C++ 1.50.1 (was 1.44.0)
Installing gRPC-Core 1.50.1 (was 1.44.0)
Removing FirebaseCoreDiagnostics
Removing FirebaseStorageInternal
Removing GoogleDataTransport
Removing Libuv-gRPC
```

Context: macOS 10.12 was the deployment target in 9.6.0 but we increased it to macOS 10.13 in Firebase 10.0.0 to match the minimum supported by Xcode 14.

If this doesn't resolve the issue, please share your `Podfile.lock` as well. Hope this helps!

**Comment by Jaycyn** [comment]
@andrewheard

Thanks for the quick response and the info - your guess was correct. The original intention was to stay at 9.6 for this test project but it appears that is not feasible. It is correct that the pods were 'out of sync' as some updated beyond what was compatible with 9.6?

I made your suggested change; then deintegrated cocoapods, deleted the podfile.lock opened the project and cleaned, ran it to ensure it was stable, quit and did a pod install. Note there is no other code in the project, just a default.

That resulted in a number of similar warnings being displayed like this (about 15 total);

> .../FirestoreTest/Pods/Pods.xcodeproj The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.

but there's also a hard error now

> xray.upbdefs.c
> '.../FirestoreTest/Pods/gRPC-Core/src/core/ext/upbdefs-generated/envoy/config/trace/v3/xray.upbdefs.c module map file '.../FirestoreTest/Pods/Headers/Private/openssl_grpc/BoringSSL-GRPC.modulemap' not found

The Podfile.lock is quite long but here are a few that were flagged with warnings (if you need the whole thing, I will edit)

>  - BoringSSL-GRPC (0.0.24):
>     - BoringSSL-GRPC/Implementation (= 0.0.24)
>     - BoringSSL-GRPC/Interface (= 0.0.24)
>  
> - GTMSessionFetcher/Core (3.1.1)
> 
> - GoogleUtilities/AppDelegateSwizzler (7.11.1):
>     - GoogleUtilities/Environment
>     - GoogleUtilities/Logger
>     - GoogleUtilities/Network
>   - GoogleUtilities/Environment (7.11.1):
>     - PromisesObjC (< 3.0, >= 1.2)
>   - GoogleUtilities/Logger (7.11.1):
>     - GoogleUtilities/Environment
>  
> - PromisesObjC (2.2.0)

here's the Firebase Section

> - FirebaseAppCheckInterop (10.11.0)
>   - FirebaseAuth (10.11.0):
>     - FirebaseAppCheckInterop (~> 10.0)
>     - FirebaseCore (~> 10.0)
>     - GoogleUtilities/AppDelegateSwizzler (~> 7.8)
>     - GoogleUtilities/Environment (~> 7.8)
>     - GTMSessionFetcher/Core (< 4.0, >= 2.1)
>   - FirebaseAuthInterop (10.11.0)
>   - FirebaseCore (10.11.0):
>     - FirebaseCoreInternal (~> 10.0)
>     - GoogleUtilities/Environment (~> 7.8)
>     - GoogleUtilities/Logger (~> 7.8)
>   - FirebaseCoreExtension (10.11.0):
>     - FirebaseCore (~> 10.0)
>   - FirebaseCoreInternal (10.11.0):
>     - "GoogleUtilities/NSData+zlib (~> 7.8)"
>   - FirebaseFirestore (10.11.0):
>     - abseil/algorithm (~> 1.20220623.0)
>     - abseil/base (~> 1.20220623.0)
>     - abseil/container/flat_hash_map (~> 1.20220623.0)
>     - abseil/memory (~> 1.20220623.0)
>     - abseil/meta (~> 1.20220623.0)
>     - abseil/strings/strings (~> 1.20220623.0)
>     - abseil/time (~> 1.20220623.0)
>     - abseil/types (~> 1.20220623.0)
>     - FirebaseCore (~> 10.0)
>     - "gRPC-C++ (~> 1.50.1)"
>     - leveldb-library (~> 1.22)
>     - nanopb (< 2.30910.0, >= 2.30908.0)
>   - FirebaseFirestoreSwift (10.11.0):
>     - FirebaseCore (~> 10.0)
>     - FirebaseCoreExtension (~> 10.0)
>     - FirebaseFirestore (~> 10.0)
>     - FirebaseSharedSwift (~> 10.0)
>   - FirebaseSharedSwift (10.11.0)
>   - FirebaseStorage (10.11.0):
>     - FirebaseAppCheckInterop (~> 10.0)
>     - FirebaseAuthInterop (~> 10.0)
>     - FirebaseCore (~> 10.0)
>     - FirebaseCoreExtension (~> 10.0)
>     - GTMSessionFetcher/Core (< 4.0, >= 2.1)

**Comment by andrewheard** [comment]
@Jaycyn Unfortunately some of those `The macOS deployment target 'MACOSX_DEPLOYMENT_TARGET' is set to 10.10, but the range of supported deployment target versions is 10.13 to 13.3.99.` warnings in transitive dependencies are expected (e.g., GTMSessionFetcher, PromisesObjC, nanopb, etc.) since those projects haven't increased their minimum deployment targets to 10.13. Generally we'd recommend ignoring those warnings but a possible workaround would be to use an approach similar to that described in this [Stack Overflow answer](https://stackoverflow.com/a/63970981).

<details>
<summary>Example Post-Install Snippet</summary>

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      if config.build_settings['MACOSX_DEPLOYMENT_TARGET'].to_f < 10.13
        config.build_settings['MACOSX_DEPLOYMENT_TARGET'] = '10.13'
      end
    end
  end
end
```
Note: the above snippet doesn't consider semantic versioning (just converts to floats) so 10.9 > 10.13 -- it works better for macOS 11+.
</details>

For the hard error,

> xray.upbdefs.c
> '.../FirestoreTest/Pods/gRPC-Core/src/core/ext/upbdefs-generated/envoy/config/trace/v3/xray.upbdefs.c module map file '.../FirestoreTest/Pods/Headers/Private/openssl_grpc/BoringSSL-GRPC.modulemap' not found

does this go away if you "Clean Build Folder..." _after_ re-opening the xcworkspace? I think the build folder is different between the xcodeproj and xcworkspace so it may not actually be a clean state if you didn't clean again after `pod install`ing and opening the workspace (Unfortunately I'm just guessing on this since I wasn't able to reproduce the modulemap error myself.)

**Comment by Jaycyn** [comment]
@andrewheard

I cleaned build before and after which had no effect.

As a further test, I am at my office and factory reset an iMac (Retina 5K, 27-inch, 2019) upgraded to Monterey 12.6.5, installed XCode and Command line tools, ensured Cocoapods and repo were all up to date.

I followed the exact process I outlined above and am receiving similar results - with a number of libraries throwing a warning but then a hard error and Build Failed

> .../FirestoreTest/Pods/gRPC-Core/src/core/ext/upbdefs-generated/envoy/config/trace/v3/zipkin.upbdefs.c
> .../FirestoreTest/Pods/gRPC-C++/src/cpp/util/time_cc.cc module map file .../FirestoreTest/Pods/Headers/Private/grpc/gRPC-Core.modulemap' not found

![Firebase Error](https://github.com/firebase/firebase-ios-sdk/assets/19558530/e0b100df-04d4-4909-8911-f047fad0b9ac)

**Comment by paulb777** [comment]
Sorry about the trouble @Jaycyn.  We're not likely to ever be able to completely eliminate the version warnings from Firebase dependencies since some of them want to continue to support Xcode 13 and earlier versions.

Regarding the hard failure, we'll likely need a reproducible use case to help since we haven't seen the issue in our testing.

**Comment by andrewheard** [comment]
Hi @Jaycyn, just trying to narrow down any other possible differences between our project setups. Is the new project you created in Objective-C or Swift?

It is also seems like you've been able to reproduce the issue on Ventura (since Xcode 14.3.1 requires it) and Monterey (the fresh iMac). Is that accurate?

Sorry for all the back and forth!

**Comment by Jaycyn** [comment]
A bit more info. Seems the issue can be narrowed to Ventura.

Starting yesterday morning, I could not build/run the project on anything.

After deintegrating, removing podfile.lock and rebuilding, the projects now build on the Monterey Macs with only a warning. It's possible something in the libraries changed yesterday, allowing the build.

I crafted a fresh project (called afirestore) following the above steps as well and it also builds on Monterey

I copied the project to the Ventura Mac and when building throws a hard error

.../afirestore/Pods/gRPC-Core/src/core/ext/upbdefs-generated/envoy/config/trace/v3/zipkin.upbdefs.c
.../afirestore/Pods/gRPC-Core/src/core/ext/upbdefs-generated/envoy/config/trace/v3/zipkin.upbdefs.c module map file '.../afirestore/Pods/Headers/Private/openssl_grpc/BoringSSL-GRPC.modulemap' not found

I then deintegrated, removed podfile.lock etc and then pod installed

That project throws the above error

Lastly, I created the project on the Ventura Mac, following the above steps and it will not build without the hard error. 

On the Monterey Macs, removing cocoapods from the project and pod installing allows it to then build

So bottom line is Monterey Builds are ok (with warnings) but Ventura builds throw hard error.

**Comment by paulb777** [comment]
Will you share the project via zip or GitHub?

**Comment by Jaycyn** [comment]
@paulb777 

Certainly. The attached zip file was built on Ventura and the podfile includes this line

`use_frameworks! :linkage => :static`

I believe the project builds and runs on Ventura as well as Monterey.

If pod deintegrate is performed (then removing podfile.lock, the xcworkspacefile etc) and that line is commented out and replaced with

`use_modular_headers!`

And pod install is run, the project will no longer build on Monterey and XCode generates an error regarding zipkin.upbdefs.c

As noted from CocoaPods if *neither* is included (which is why some clarity may be needed in the docs)

[!] The following Swift pods cannot yet be integrated as static libraries:

> The Swift pod `FirebaseCoreInternal` depends upon `GoogleUtilities`, which does not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.

So I seem to be getting really intermittent results trying to duplicate the issue - these are all being tested on 100% clean Macs, factory fresh status using Monterey on one Ventura on the other.

[afirestore.zip](https://github.com/firebase/firebase-ios-sdk/files/11969272/afirestore.zip)

**Comment by andrewheard** [comment]
Hi @Jaycyn, sorry I missed the "Can also `use_modular_headers!`?" question in your Podfile comment. Firebase requires the use of frameworks as of version 9.0 when we introduced some Swift pods ([Release Note](https://firebase.google.com/support/release-notes/ios#version_900_-_may_3_2022)) so `use_modular_headers!` won't work:
> Breaking change: Podfiles must include use_frameworks! or use_frameworks! :linkage => :static.

I'll add a note to our installation [docs](https://firebase.google.com/docs/ios/installation-methods#cocoapods) since it isn't mentioned there.

**Comment by paulb777** [comment]
@Jaycyn Thanks for your patience in helping us track down the documentation issue! I'll close but feel free to continue the discussion.
